### DISCUSSION TITLE
English dialect selection - Automatic or add detect_language=true?

### DISCUSSION THREAD

**Discussion by AntonBuyan** [deepgram/community#799] [General help] [answered]
Question - our users are a mix of english speakers (GB, US, India, etc) and we use model=nova-2.

To get the best transcription quality, should I specify "en-US"  and also enable language detection for the best variant (detect_language=true). Or should I just use language=en and Deepgram will select the best dialect automatically?

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Reply by SandraRodgers** [reply]
**SEE UPDATED ANSWER IN LATER COMMENT**

If your users are a mix of English speakers from different regions and you're unsure of the specific dialect in each audio, it would be best to use `detect_language=true` without specifying a language parameter. This way, Deepgram will automatically detect and use the most appropriate English dialect for transcription. 

While adding `language=en` is generally harmless, it might add a layer of complexity that can be avoided by solely relying on the `detect_language=true` feature. In other words, it would be unnecessary and you should get the results you are looking for just with `language-detect`. If that's not working well for you, then I would be interested to know that!

**Reply by AntonBuyan** [reply]
to restrict detection only to English and auto-select the best dialect, should we use "detect_language=en" ? cc @SandraRodgers  (we don't want to fully open up to other languages, e.g. spanish, as that is a bigger product change)

**Comment by SandraRodgers** [comment, marked as answer]
Hi @AntonBuyan

I am going to change my answer from the above. I have found out some new information. Apologies for the first inaccurate answer.

The best thing for you to do if you are using Nova-2 is to just set language=en. This will ensure that the model sticks to English, and the great thing about Nova-2 is that it will automatically detect the dialect of English and transcribe it. In previous models, we suggested using detect_language due to Deepgram having standalone different English dialect models, but now the Nova-2 model on its own will be able to detect the dialect for you. You do not need to use detect_language

And as an added note for anyone who comes across this who wants to limit the languages that are detected, such as telling Deepgram to only detect Spanish or French, they could actually set detect_language=es&detect_language=fr. This method could be done for any of the languages that language detection supports (so not all of Deepgram's available languages). However, this would not work for the English dialects. You should use language=en with Nova-2 to do this.
